File: java-workshop/part-1-powermock/src/main/java/service/ShippingService.java
package service;

import domain.Order;
import domain.ShippingConfirmation;
import domain.ShippingItem;
import domain.ShippingRequest;
import infrastructure.ShippingApi;

import java.util.stream.Collectors;

/**
 * Creates ShippingApi with new - requires PowerMock to mock constructor.
 */
public class ShippingService {

    public ShippingConfirmation createShipment(Order order) {
        ShippingApi api = new ShippingApi();
        ShippingRequest request = new ShippingRequest(
            order.id(),
            order.customerId(),
            order.items().stream()
                .map(item -> new ShippingItem(item.productId(), item.quantity()))
                .collect(Collectors.toList())
        );
        return api.createShipment(request);
    }

    public boolean cancelShipment(String trackingNumber) {
        ShippingApi api = new ShippingApi();
        return api.cancelShipment(trackingNumber);
    }
}

File: csharp-workshop/Part1.StaticDeps/Service/ShippingService.cs
using Part1.StaticDeps.Domain;
using Part1.StaticDeps.Infrastructure;

namespace Part1.StaticDeps.Service;

/// <summary>
/// Creates ShippingApi with new — requires constructor mocking (JustMock/Typemock) to test.
/// Java equivalent requires PowerMock to mock the constructor.
/// </summary>
public class ShippingService
{
    public ShippingConfirmation CreateShipment(Order order)
    {
        var api = new ShippingApi();
        var request = new ShippingRequest(
            order.Id,
            order.CustomerId,
            order.Items
                .Select(item => new ShippingItem(item.ProductId, item.Quantity))
                .ToList()
                .AsReadOnly());
        return api.CreateShipment(request);
    }

    public bool CancelShipment(string trackingNumber)
    {
        var api = new ShippingApi();
        return api.CancelShipment(trackingNumber);
    }
}
